File: test/unit-testing/circle.c
#define DWG_TYPE DWG_TYPE_CIRCLE
#include "common.c"

void
api_process (dwg_object *obj)
{
  int error;
  double radius, thickness;
  dwg_point_3d center, extrusion;

  dwg_ent_circle *circle = dwg_object_to_CIRCLE (obj);

  CHK_ENTITY_3RD_W_OLD (circle, CIRCLE, center);
  CHK_ENTITY_TYPE_W_OLD (circle, CIRCLE, radius, BD);
  CHK_ENTITY_TYPE_W_OLD (circle, CIRCLE, thickness, BD);
  CHK_ENTITY_3RD_W_OLD (circle, CIRCLE, extrusion);
}

File: bindings/javascript/src/types/entities/circle.ts
import { DwgPoint3D } from '../common'
import { DwgEntity } from './entity'

export interface DwgCircleEntity extends DwgEntity {
  type: 'CIRCLE'
  thickness: number
  center: DwgPoint3D
  radius: number
  extrusionDirection: DwgPoint3D
}
